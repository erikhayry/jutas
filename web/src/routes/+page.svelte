<script lang="ts">
    import {resolve} from '$app/paths';
    import {getComics} from "../utils/getComics";
</script>

<ul>
    {#each getComics() as comic (comic)}
        <li>
            <a href={resolve(`/${comic.slug}`)}>{comic.slug}</a>
        </li>
    {/each}
</ul>
