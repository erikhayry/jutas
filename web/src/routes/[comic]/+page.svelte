<script lang="ts">
    import type { PageProps } from './$types';

    let { data }: PageProps = $props();
</script>


<h1>{data.title}</h1>

<ul>

{#each data.pages as page, index}
    <li>
        <a href={`/${data.slug}/${index}`}>{page.title}</a>
    </li>
{/each}
</ul>


<a href="/">Back</a>