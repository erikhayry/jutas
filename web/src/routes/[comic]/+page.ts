import {error} from '@sveltejs/kit';
import {getComic} from "./utils/getComic";

export const load = ({params}: { params: { comic: string } }) => {
    const comic = getComic(params.comic)

    if (comic) {
        return comic
    }

    error(404, 'Not found');
};