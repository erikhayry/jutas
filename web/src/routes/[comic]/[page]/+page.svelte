<script lang="ts">
    import {resolve} from '$app/paths';
    import Img from "../../../components/PanelImage/PanelImage.svelte";

    let {data} = $props();
    const {slug, page, title} = data
</script>

<h1>{title}</h1>

{#each page.panels as panel (panel)}
    <Img alt="TODO" comic={slug} id={panel.id}/>
{/each}

<a href={resolve(`/${slug}`)}>Back</a>