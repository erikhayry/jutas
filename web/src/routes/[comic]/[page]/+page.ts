import {error} from '@sveltejs/kit';
import {getPage} from "./utils/getPage";
import type {Page} from "jutas-types";

export const load = ({params: {comic: slug, page: pageIndex}}: { params: { comic: string, page: string } }): {
    page: Page,
    slug: string
    title: string

} => {
    const page = getPage(slug, parseInt(pageIndex))

    if (page) {
        return {
            title: `${slug} | ${parseInt(pageIndex) + 1}`,
            page,
            slug,
        }
    }

    error(404, 'Not found');
};