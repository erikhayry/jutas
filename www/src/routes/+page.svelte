<script lang="ts">
    import data from '$lib/assets/data.json';
</script>

<ul>
{#each data as comic}
    <li>
        <a href={`/${comic.slug}`}>{comic.title}</a>
    </li>
{/each}
</ul>
