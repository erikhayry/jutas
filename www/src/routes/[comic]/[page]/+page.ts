import { error } from '@sveltejs/kit';
import type { PageLoad } from './$types';
import data from '$lib/assets/data.json';


export const load: PageLoad = ({ params }) => {
    const comicPage = data.find(({ slug}) => slug === params.comic)?.pages?.[Number.parseInt(params.page)]

    if (comicPage) {
        return comicPage
    }

    error(404, 'Not found');
};