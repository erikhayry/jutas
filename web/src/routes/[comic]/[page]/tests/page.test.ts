import {describe, expect, it} from 'vitest';
import {load} from "../+page";
import {COMIC_MOCK, PAGE_INDEX_MOCK, PAGE_MOCK} from "../../../../test/mock";


describe('Comic page load', () => {
    it('should load comic page if available', async () => {
        expect(load({params: {comic: COMIC_MOCK.slug, page: PAGE_INDEX_MOCK.toString()}})).toEqual({
            page: PAGE_MOCK,
            slug: COMIC_MOCK.slug,
            title: `${COMIC_MOCK.slug} | ${PAGE_INDEX_MOCK + 1}`
        })
    });

    it('should load error if not available', async () => {
        try {
            load({params: {comic: COMIC_MOCK.slug, page: '99'}})
        } catch (error) {
            expect(error).toMatchSnapshot()
        }
    });
});
