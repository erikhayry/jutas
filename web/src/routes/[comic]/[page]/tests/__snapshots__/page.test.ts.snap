// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Comic page load > should load error if not available 1`] = `
HttpError {
  "body": {
    "message": "Not found",
  },
  "status": 404,
}
`;

exports[`Comic page load should load error if not available 1`] = `
HttpError {
  "body": {
    "message": "Not found",
  },
  "status": 404,
}
`;
